.github/workflows/pr_tests.yml

```diff
@@ -0,0 +1,35 @@
+name: Run non-slow tests
+
+on:
+  pull_request:
+    branches:
+      - main
+
+env:
+  HF_HOME: /mnt/cache
+  OMP_NUM_THREADS: 8
+  MKL_NUM_THREADS: 8
+  PYTEST_TIMEOUT: 60
+
+jobs:
+  run_tests_cpu:
+    name: Diffusers tests
+    runs-on: [ self-hosted, docker-gpu ]
+    container:
+      image: nvcr.io/nvidia/pytorch:22.07-py3
```

Comment:
wonder whether it might be easier to here just use `pip install torch`? Like in https://github.com/huggingface/transformers/blob/62ceb4d661ce644ee9377ac8053cbb9afa737125/.circleci/config.yml#L126

Reply:
Or maybe even easier we define a `pip install . [dev]` where we could easily add 'jax' in the future? 
https://github.com/huggingface/transformers/blob/62ceb4d661ce644ee9377ac8053cbb9afa737125/.github/workflows/add-model-like.yml#L37

Reply:
But no expert here, just wonder if `nvidia/pytorch` docker is the best environment to test pytorch (+ potentially jax in the future) for CPU

Reply:
Also maybe I'd set up a new CPU machine (they're cheap) so that we can easier disentangle CPU and GPU?
Overall I'm no testing expert here, so just a couple of thoughts - feel free to proceed however you like best (and how it's most efficient for you to debug things going forward)

Reply:
Agreed, let's start with a bare minimum "python:3.7" image and set up the frameworks on top for better environment control.
Ideally we'll need a custom image with torch and flax preinstalled (to avoid waiting for them to install on each run), but for now installing them from scratch is good enough.

Will test it now

Reply:
Ok, I take that back, our cloud network speeds are crazy fast, will use `python:3.7` for CPU tests from now on :tada:

Reply:
Are we testing on CPU here, as we use GPU VMs ?

Reply:
I guess `main` here means  the opened PR want to be merge into `main`?

Reply:
Yes, correct